document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -445,6 +429,58 @@ public boolean bulkUpsert(Map<Key, Document> documents) {
     return false;
   }
 
+  private int[] bulkUpsertImpl(Map<Key, Document> documents) throws SQLException, IOException {
+    PreparedStatement preparedStatement = client
+        .prepareStatement(getUpsertSQL(), Statement.RETURN_GENERATED_KEYS);
+    for (Map.Entry<Key, Document> entry : documents.entrySet()) {
+
+      Key key = entry.getKey();
+      String jsonString = prepareUpsertDocument(key, entry.getValue());
+
+      preparedStatement.setString(1, key.toString());
+      preparedStatement.setString(2, jsonString);
+      preparedStatement.setString(3, jsonString);
+
+      preparedStatement.addBatch();
+    }
+
+    return preparedStatement.executeBatch();
+  }
+
+  @Override
+  public Iterator<Document> returnAndBulkUpsert(Map<Key, Document> documents) throws IOException {
+    try {
+      String collect = documents.keySet().stream()
+          .map(val -> "'" + val.toString() + "'")
+          .collect(Collectors.joining(", "));
+
+      String space = " ";
+      String query = new StringBuilder("SELECT * FROM")
```

Comment:
This seems like a shortcut to SQL injection, no?

Reply:
Hmm.. Thanks for pointing. Let me check and think more..

Reply:
I think the SQLi is being fixed as a part of https://github.com/hypertrace/document-store/pull/16/files
I'll merge this and will update the other PR and will fix SQLi as a part of that. If the other PR is merged before mine, i can update and fix it.